Include notes into linter messages

ROpdebee opened the pull request:
Options like --warn-unused-ignores and --warn-redundant-casts output notes, which would otherwise not be displayed

FichteFoll commented:
How is the UX with the added note lines? I know it was bad in SL3 but haven't considered looking into how it might integrate in SL4 with the improved hover popup and all.
Are notes still emitted *before* the warning/error they annotate?

ROpdebee commented:
A sample:
<img width="730" alt="screenshot 2019-01-26 at 21 17 11" src="https://user-images.githubusercontent.com/15186467/51792332-c46d4680-21af-11e9-9ae9-ea1bfa7751d1.png">

There also shouldn't be any notes generated (like `note: In function "testfunction":`) since `--hide-error-context` is passed to mypy.

IMO the redundant cast note shouldn't really be displayed as a warning, more as a hint, but I don't think SL4 has support for message types other than error or warning.

E: Although this might be an issue...:
<img width="816" alt="screenshot 2019-01-26 at 21 30 24" src="https://user-images.githubusercontent.com/15186467/51792472-a9033b00-21b1-11e9-98dd-fba1291e859e.png">
It can be suppressed by the `--ignore-missing-imports` flag, but that prevents the error from showing too, which may be undesirable if you want to create stubs for 3rd party modules/packages that are not in the typeshed.

fredcallaway commented:
Unclear to me why this is failing the Travis build... but looks good otherwise. If the current behavior is to not show any explanation for the redundant cast warning, then this is definitely an improvement.

By the way, I really like your color scheme! Can you share it?

ROpdebee commented:
Travis build is failing because the latest version of flake8 linter requires Python >= 3.4. I've updated the config to use 3.6 since that's what seems to be what most SL plugins are using now.

Theme is [DA UI](https://packagecontrol.io/packages/DA%20UI) with the DA Minimal skin and the DA Dark color scheme. Here's a full screenshot: 
<img width="1680" alt="screenshot 2019-01-27 at 21 55 45" src="https://user-images.githubusercontent.com/15186467/51806828-0dd89700-227f-11e9-9cec-e73424f09944.png">
But now that you mentioned it, I changed it to the DA Elegant skin: 
<img width="1680" alt="screenshot 2019-01-27 at 22 02 52" src="https://user-images.githubusercontent.com/15186467/51806856-51330580-227f-11e9-8aa1-4c7f351a37ee.png">

FichteFoll commented:
Thanks for the screenshots and clarification.

Included in 1.3.6.
